src/test/integration/integration-tests.py

```diff
@@ -0,0 +1,290 @@
+        validate_bad_list(self, "instances", None,
+                          valid_id,
+                          "At least one event instance is required",
+                          )
+
+        # Start date occurs after end date
+        validate_bad_list(self, "instances", [{
+            "start": "2018-12-02T21:30:00Z",
+            "end": "2018-12-02T21:15:00Z"
+        }],
+            valid_id,
+            "start time of an instance must occur before the end time",
+        )
+
+        # Invalid campusID
+        event_body["data"]["attributes"]["campusID"] = "badCampus"
+        bad_campus = utils.put_event(valid_id, event_body)
+        validate_response(self, bad_campus, 400, message="campusID")
+        event_body["data"]["attributes"]["campusID"] = None
+
+        # Invalid audienceIDs
+        validate_bad_list(self, "audienceIDs", ["badAudience"], valid_id)
+
+        # Invalid countyIDs
+        validate_bad_list(self, "countyIDs", ["badCounty"], valid_id)
+
+        # Invalid departmentIDs
+        validate_bad_list(self, "departmentIDs", ["badDepartment"], valid_id)
+
+        # Invalid eventTopicIDs
+        validate_bad_list(self, "eventTopicIDs", ["badTopic"], valid_id)
+
+        # Invalid eventTypeIDs
+        validate_bad_list(self, "eventTypeIDs", ["badType"], valid_id)
+
+        # Hashtag field contains "#"
+        invalid_event_attribute(self, "hashtag", "#", valid_id)
+
+        # No title
+        invalid_event_attribute(self, "title", None, valid_id)
+
+        # No description
+        invalid_event_attribute(self, "description", None, valid_id)
+
+        # No contactName
+        invalid_event_attribute(self, "contactName", None, valid_id)
+
+        # No contactEmail
+        invalid_event_attribute(self, "contactEmail", None, valid_id)
+
+        # Clean up test event
+        utils.delete_event(valid_id)
+
+    # Test GET /calendar/event-types and GET /calendar/event-types/{id}
+    def test_event_types(self):
+        test_get(self, "event-types")
+
+    # Test GET /calendar/event-topics and GET /calendar/event-topics/{id}
+    def test_event_topics(self):
+        test_get(self, "event-topics")
+
+    # Test GET /calendar/counties and GET /calendar/counties/{id}
+    def test_counties(self):
+        test_get(self, "counties")
+
+    # Test GET /calendar/locations and GET /calendar/locations/{id}
+    def test_locations(self):
+        test_get(self, "locations", is_paginated=True)
+
+    # Test GET /calendar/campuses and GET /calendar/campuses/{id}
+    def test_campuses(self):
+        test_get(self, "campuses", is_paginated=True)
+
+    # Test GET /calendar/departments and GET /calendar/departments/{id}
+    def test_departments(self):
+        test_get(self, "departments", is_paginated=True)
+
+    # Test GET /calendar/audiences and GET /calendar/audiences/{id}
+    def test_audiences(self):
+        test_get(self, "audiences")
+
+    # Test GET /calendar/feed
+    def test_feed(self):
```

Comment:
Could you do some more testing with the feed endpoint, like parsing the response to make sure its a proper CSV?